hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/processor/GroupP.java

```diff
@@ -82,4 +81,26 @@ protected boolean tryProcess(int ordinal, @Nonnull Object item) {
     public boolean complete() {
         return emitFromTraverser(resultTraverser);
     }
+
+    private class ResultTraverser implements Traverser<OUT> {
+        private final BiFunction<? super K, ? super R, OUT> mapToOutputFn;
+        private Iterator<Entry<K, A>> iter;
+
+        ResultTraverser(BiFunction<? super K, ? super R, OUT> mapToOutputFn) {
+            this.mapToOutputFn = mapToOutputFn;
+        }
+
+        @Override
+        public OUT next() {
+            if (iter == null) {
+                iter = keyToAcc.entrySet().iterator();
+            }
+            if (!iter.hasNext()) {
+                return null;
+            }
+            Entry<K, A> e = iter.next();
+            iter.remove();
+            return mapToOutputFn.apply(e.getKey(), aggrOp.finishFn().apply(e.getValue()));
```

Comment:
We should handle map-to-null case here, it was handled before in `Traverser.map`. We should also add test for that.